draft-ietf-httpbis-http2bis.xml

```diff
@@ -2847,10 +2847,42 @@
             registered HTTP fields, see the "Hypertext Transfer Protocol (HTTP) Field Name Registry" registry maintained at <eref target="https://www.iana.org/assignments/http-fields/"/>.
           </t>
           <t>
-            Just as in HTTP/1.x, field names are strings of ASCII characters that are
-            compared in a case-insensitive fashion. Field names MUST be converted
-            to lowercase prior to their encoding in HTTP/2. A request or response containing
-            uppercase header field names MUST be treated as <xref target="malformed">malformed</xref>.
+            Field names are strings of ASCII characters that are compared in a case-insensitive
+            fashion. Field names MUST be converted to lowercase when constructing a HTTP/2
+            message. A request or response containing an uppercase character ('A' to 'Z', ASCII 0x41
+            to 0x5a) in a field name MUST be treated as <xref target="malformed">malformed</xref>.
+          </t>
+          <t>
+            HPACK is capable of carrying field names or values that are not valid in HTTP.  Though
+            HPACK can carry any octet, fields are not valid in the following cases:
+          </t>
+          <ul>
+            <li>
+              A field name MUST NOT contain characters in the range 0x00-0x20 (inclusive) or 0x7F
+              (ASCII DEL).  This includes all ASCII control characters, plus ASCII SP (0x20).
+            </li>
+            <li>
+              With the exception of <xref target="PseudoHeaderFields">pseudo-header fields</xref>,
+              which have a name that starts with a single colon, field names MUST NOT include a
+              colon (ASCII COLON, 0x3a).
+            </li>
+            <li>
+              A field value MUST NOT contain the zero value (ASCII NUL, 0x0), line feed (ASCII LF,
```

Comment:
This seems to repeat a requirement from above.

Reply:
FWIW, it's also not clear to me why we would allow whitespace *inside* the field name...

Reply:
Where does it say those things?

Reply:
It does not. I was confused because of the statement about field *values*. Sorry.